src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -165,6 +172,106 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
     }
 
     /**
+     * Information a bout a resource
+     * - How many are needed to finish the build
+     * - How many are available to the builder
+     * - How many are in the player's inventory (client side only)
+     */
+    public static class BuildingBuilderResource
+    {
+        /**
+         * Availability status of the resource.
+         * according to the builder's chest, inventory and the player's inventory
+         */
+        public enum RessourceAvailability
+        {
+            NOT_NEEDED,
+            DONT_HAVE,
+            NEED_MORE,
+            HAVE_ENOUGHT
+        }
+
+        private int amountAvailable;
+        private int amountPlayer;
+        final private String name;
+        final private ItemStack itemStack;
```

Comment:
In general you have Ex: `private static final String`, or `Visibility [static] [final] Type`